library/ssl_tls13_server.c

```diff
@@ -1533,6 +1533,12 @@ static int ssl_tls13_parse_client_hello(mbedtls_ssl_context *ssl,
         unsigned int extension_type;
         size_t extension_data_len;
         const unsigned char *extension_data_end;
+        uint32_t allowed_exts = MBEDTLS_SSL_TLS1_3_ALLOWED_EXTS_OF_CH;
+
+        if (ssl->handshake->hello_retry_request_count > 0) {
+            /* Do not accept early data extension in 2nd ClientHello */
```

Comment:
Is there a test that verifies we don't accept early data in 2nd CH?

Reply:
No we do not have such test. It would involve a non-compliant client and is not easy to set up. We consider such tests as low priority for the time being.